tests/codegen/llvm/avg_cast_loop.ll

```diff
@@ -24,31 +24,35 @@ define i64 @kprobe_f_1(ptr %0) section "s_kprobe_f_1" !dbg !72 {
 entry:
   %avg_struct = alloca %avg_stas_val, align 8
   %"@x_key" = alloca i64, align 8
+  %get_cpu_id = call i64 inttoptr (i64 8 to ptr)()
+  %1 = load i64, ptr @max_cpu_id, align 8
+  %2 = icmp ule i64 %get_cpu_id, %1
+  %3 = select i1 %2, i64 %get_cpu_id, i64 %1
```

Comment:
It's a bit unfortunate this is still generated even though cpuid is not used from main prog. It's being generated in all progs b/c the map cb needs it. I'm not sure it's worth the complexity to fix this case, as basically all non-trivial scripts will end up using printf() or tuples() or map accesses (etc) in main prog. Just calling it out here.

Reply:
Yea I saw that too.

We could implement a Probe/Subprog -> "does this need cpu_id caching" in ResourceAnalyser. But I agree...feels like a lot of complexity.

The good news is LLVM optimizer realizes this and only keeps the get_cpu_id instruction in the optimized codegen (because it doesn't know if the BPF helper has side effects). But this instruction is cheap.
```
define noundef i64 @kprobe_f_1(ptr nocapture readnone %0) local_unnamed_addr section "s_kprobe_f_1" !dbg !72 {
entry:
  %avg_struct = alloca %avg_stas_val, align 8
  %"@x_key" = alloca i64, align 8
  %get_cpu_id = tail call i64 inttoptr (i64 8 to ptr)()
  call void @llvm.lifetime.start.p0(i64 -1, ptr nonnull %"@x_key")
  store i64 1, ptr %"@x_key", align 8
  %lookup_elem = call ptr inttoptr (i64 1 to ptr)(ptr nonnull @AT_x, ptr nonnull %"@x_key")
  %lookup_cond.not = icmp eq ptr %lookup_elem, null
  br i1 %lookup_cond.not, label %lookup_failure, label %lookup_success
```